docs/tests/conditionals.rst

```diff
@@ -0,0 +1,188 @@
+
+not_if
+~~~~~~
+
+The ``not_if`` section of the pavilion test config behaves exactly
+like the ``only_if`` section except for one main difference. The
+test will be assigned the status of ``SKIPPED`` if at least one
+conditional match is found. Let's take a look at an example:
+
+.. code:: yaml
+
+    basic_test:
+        not_if:
+            sys_os: ['windows']
+        run:
+            cmds:
+                ...
+
+    In this example the 'not_if' directive is followed by a dictionary
+    with the key 'sys_os' and the value of 'windows'. Pavilion will resolve
+    the key and check what the operating system on the machine is. If
+    sys_os resolves to 'windows' then the 'not_if' conditions has a match
+    and the test will be skipped.
+
+Conditional statements can also parse dictionaries with multiple keys
+and values, let's look at a more complicated test.
+
+.. code:: yaml
+
+    advanced_test:
+        not_if:
+            user: ['calvin', 'nick']
+            sys_os: ['linux']
+        run:
+           cmds:
+               ...
+
+    In this example Pavilion will need to resolve two keys, user,
+    sys_os. When the keys are resolved they will be compared
+    to the dictionary values supplied to them. It is important to note
+    for 'not_if', it only takes 1 match to skip the test. In this case
+    if the user is either 'calvin' or 'nick', or the operating system
+    is 'linux', the test will be skip.
+
+Variables
+~~~~~~~~~
+
+Throughout this documentation variables are synonymous with keys. Keys
+being the literal dictionary key supplied after calling ``not_if`` or
+``only_if``. There are multiple types of variables supported in Pavilion
+and for detailed documentation on what variables to use, and how to create
+you own variables see `Variables <variables.html>`__.
+
+Tips & Tricks
+~~~~~~~~~~~~~
+Below contains useful bits of information that can help users customize
+their conditional statements.
+
+Deferred Variables
+^^^^^^^^^^^^^^^^^^
+
+Deferred Variables in Pavilion are variables that cannot be resolved
+until after an allocation, for example host_arch and number of nodes.
+Conditionals handle this by evaluating the test twice, on the second
+attempt all deferred variables will have been resolved. The test will
+then be properly assigned to ``SKIPPED`` if needed.
+
+Mixed Use
+^^^^^^^^^
+The ``not_if`` and ``only_if`` directives can be used in conjunction with
+one another. The easiest way to see this is by example:
+
+.. code:: yaml
+
+    mixed_use_test:
+        only_if:
+            user: ['francine', 'paul']
+            sys_os: ['linux']
+        not_if:
+            weekday: ['saturday', 'sunday']
+            sys_arch: ['aarch64']
+        run:
+            cmds:
+                ...
+
+Regex
+^^^^^
+
+All conditional statement directives have dictionaries that follow them. The
+values following the keys in the dictionaries are all interpreted as regex
+patterns. Let's look at the following example:
```

Comment:
So much re-explanation...

Just say every value is a regex that much fully match the _whole_ key. A well commented example is appropriate. The example should also point out that 'calvin' is a regex too, and won't match 'calvinsmith'.